include/deal.II/lac/la_parallel_vector.h

```diff
@@ -1082,8 +1082,12 @@ namespace LinearAlgebra
 
       /**
        * Pointer to the array of local elements of this vector.
+       *
+       * Because we allocate these arrays via Utilities::System::posix_memalign,
+       * we need to use a custom deleter for this object that does not call
+       * <code>delete[]</code>, but instead calls @p free().
        */
-      std::unique_ptr<Number[]> values;
+      std::unique_ptr<Number[], void (*)(void *)> values;
```

Comment:
I would prefer to use `decltype(free)*` here so we already know from the code that we want to call `free` here.

Reply:
Nice idea - done!